dll/win32/shell32/CMakeLists.txt

```diff
@@ -117,3 +117,4 @@ add_importlibs(shell32 advapi32 gdi32 user32 comctl32 comdlg32 shlwapi msvcrt ke
 add_dependencies(shell32 stdole2) # shell32_shldisp.tlb needs stdole2.tlb
 add_pch(shell32 precomp.h "${PCH_SKIP_SOURCE}")
 add_cd_file(TARGET shell32 DESTINATION reactos/system32 FOR all)
+target_compile_definitions(shell32 PRIVATE NTDDI_VERSION=NTDDI_WS03SP1)
```

Comment:
OK at the beginning I see we are going to mostly use `_WIN32_WINNT` as version check in shell32, so please remove that NTDDI thing and let's use `_WIN32_WINNT` in the ifdef we talked about in the PR. See the suggestions below (also the "Vista+" comment is unnecessary, as the preprocessor command precisely says this fact.)